### DISCUSSION TITLE
how to create traits for textcomponent and label component?

### DISCUSSION THREAD

**Discussion by senthil177cse** [GrapesJS/grapesjs#4331] [Q&A]
i have tried to add new trait of text and label component but it is not adding... but same method i have tried for image and input component its added? pls check below code and explain why its not adding or anything wrong pls correct me?

this.editor.Components.addType('text', {
      isComponent: el => el.tagName == 'p',
      model: {
        defaults: {
          traits: [
            {
              type:'checkbox',
              name: 'visibility'
            }
          ]
        }
      }
    });
    
    this.editor.DomComponents.addType('textnode', {
      isComponent: el => el.tagName == 'label',
      model: {
        defaults: {
          traits: [
            {
              type:'checkbox',
              name: 'visibility'
            }
          ]
        }
      }
    });

**Comment by ronaldohoch** [comment]
Hello, have you checked the forms plugin? Many of features i have to add in my project i found in his plugins.

https://github.com/artf/grapesjs-plugin-forms/blob/master/src/components.js#L299
